packages/nimble-components/src/rich-text-mention/base/index.ts

```diff
@@ -53,40 +52,64 @@ export abstract class RichTextMention<
 
     /** @internal */
     @observable
-    public mappings: Mapping<unknown>[] = [];
+    public mappingElements: Mapping<unknown>[] = [];
+
+    /**
+     * @public
+     * Returns hrefs for existing mentions in the editor/viewer.
+     */
+    public abstract getMentionedHrefs(): string[];
 
+    /**
+     * @public
+     */
     public checkValidity(): boolean {
         return this.mentionInternals.validConfiguration;
     }
 
+    /**
+     * @public
+     */
     public get validity(): RichTextMentionValidity {
         return this.validator.getValidity();
     }
 
+    /**
+     * @internal
+     */
+    public emitMentionUpdate(filter: string): void {
+        const mentionUpdateEventDetails: MentionUpdateEventDetail = {
+            filter
+        };
+        this.$emit('mention-update', mentionUpdateEventDetails);
+    }
+
     /**
      * @internal
      */
     public handleChange(source: unknown, args: unknown): void {
         if (source instanceof Mapping && typeof args === 'string') {
-            this.updateMentionConfig();
+            this.updateMappingConfigs();
         }
     }
 
     protected abstract createValidator(): TValidator;
 
     protected abstract getMentionInternalsOptions(): MentionInternalsOptions;
 
-    protected abstract createMentionConfig(
-        mappingConfigs: MappingConfigs
-    ): TMentionConfig;
+    protected abstract createMentionConfig(): MentionConfig;
 
     protected abstract createMappingConfig(
         mapping: Mapping<unknown>
     ): MappingConfig;
 
+    protected get richTextParent(): RichText {
+        return this.parentElement as RichText;
+    }
```

Comment:
If we can get rid of getMentionedHrefs than we can remove this too

Reply:
Removed this as `getMentionedHrefs()` get removed from the base.